From ba2d710e1742f385ca9ce56896c9d381e48a4c33 Mon Sep 17 00:00:00 2001
From: Ryan Walklin <ryan@testtoast.com>
Date: Wed, 22 May 2024 21:13:34 +1200
Subject: [PATCH] arm64: dts: allwinner: h616: add PWM node

The Allwinner H616 and variants have a PWM controller with 6 channels.

Add a device tree node for the controller.

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
---
 arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
index b2474f8a5cba..082c71068af4 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
@@ -231,6 +231,20 @@ ccu: clock@3001000 {
 			#reset-cells = <1>;
 		};
 
+		pwm: pwm@300a000 {
+			compatible = "allwinner,sun50i-h616-pwm";
+			reg = <0x0300a000 0x400>;
+			clocks = <&ccu CLK_BUS_PWM>,
+				 <&osc24M>,
+				 <&ccu CLK_APB1>;
+			clock-names = "bus", "hosc", "apb";
+			resets = <&ccu RST_BUS_PWM>;
+			#pwm-cells = <0x3>;
+			status = "disabled";
+
+			allwinner,pwm-channels = <6>;
+		};
+
 		dma: dma-controller@3002000 {
 			compatible = "allwinner,sun50i-h616-dma",
 				     "allwinner,sun50i-a100-dma";
-- 
2.43.4
